external/llvm-project/mlir/lib/Conversion/AsyncToGPU/AsyncToGPU.cpp

```diff
@@ -82,9 +83,8 @@ static Optional<DictionaryAttr> getGPUTarget(async::LaunchOp op) {
 //===----------------------------------------------------------------------===//
 
 namespace {
-class LaunchOpConversion : public OpConversionPattern<async::LaunchOp> {
-public:
-  using OpConversionPattern::OpConversionPattern;
+struct LaunchRewritePattern : public OpRewritePattern<async::LaunchOp> {
```

Comment:
Why not a conversion pattern?

Reply:
An async.launch doesn't necessarily convert to gpu.launch, only if it has a gpu target implementation.

The new conversion in async-to-async-runtime will convert any async.launch's that remain.